build/scripts/example-version-checker/version.go

```diff
@@ -0,0 +1,104 @@
+package main
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+const examplesDir = "examples"
+
+var excludedPatterns = [...]string{"*.md", "*.yaml", "OWNERS", ".gitignore"}
+
+func dirIsExample(dirName string) bool {
+	makefileName := fmt.Sprintf("%s/Makefile", dirName)
+	if _, err := os.Stat(makefileName); err == nil {
+		return true
+	} else {
+		return false
+	}
+}
+
+func getAllExampleNames() []string {
+	dirNames := make([]string, 0)
+
+	baseDir := fmt.Sprintf("%s%s", rootDir, examplesDir)
+
+	entries, err := os.ReadDir(baseDir)
+	if err != nil {
+		log.Fatalf("Could not open directory: %v", err)
+	}
+
+	for _, entry := range entries {
+		name := entry.Name()
+		path := fmt.Sprintf("%s/%s", baseDir, entry.Name())
+		if dirIsExample(path) {
+			dirNames = append(dirNames, name)
+		}
+	}
+	return dirNames
+}
+
+func getVersionFromMakefile(contents string) (string, error) {
+	scanner := bufio.NewScanner(strings.NewReader(contents))
+	for scanner.Scan() {
+		line := scanner.Text()
+		if lineContainsVersion(line) {
+			return getVersionFromLine(line), nil
+		}
+	}
+
+	if err := scanner.Err(); err != nil {
+		log.Fatalf("Could not get version: %v", err)
+	}
+
+	return "", errors.New("no version could be found")
+}
+
+func lineContainsVersion(line string) bool {
+	return strings.HasPrefix(line, "version")
```

Comment:
nit: better to make it stricter i.e. `strings.HasPrefix(line, "version :=")

Reply:
Hi. 👋 

Do you want me to action this change now?

You appear to be operating on this branch, and I have no desire to interfere. 😄

Reply:
Yeah feel free to go ahead and make the change. Thanks